src/main/java/frc/robot/commands/Drive.java

```diff
@@ -34,23 +47,50 @@ public Drive(Drivetrain subDrivetrain, DoubleSupplier xAxis, DoubleSupplier yAxi
 
   @Override
   public void initialize() {
+    northYaw = constField.isRedAlliance() ? Units.Degrees.of(180) : Units.Degrees.of(0);
+    redAllianceMultiplier = constField.isRedAlliance() ? -1 : 1;
   }
 
   @Override
   public void execute() {
+    if (slowMode.getAsBoolean()) {
+      slowMultiplier = prefDrivetrain.slowModeMultiplier.getValue();
+    } else {
+      slowMultiplier = 1;
+    }
+
     // Get Joystick inputs
-    double xVelocity = xAxis.getAsDouble() * Units.Meters.convertFrom(prefDrivetrain.driveSpeed.getValue(), Units.Feet);
-    double yVelocity = -yAxis.getAsDouble()
+    double transMultiplier = slowMultiplier * redAllianceMultiplier
         * Units.Meters.convertFrom(prefDrivetrain.driveSpeed.getValue(), Units.Feet);
+    double xVelocity = (xAxis.getAsDouble() * transMultiplier);
+    double yVelocity = (-yAxis.getAsDouble() * transMultiplier);
+
     double rVelocity = -rotationAxis.getAsDouble()
         * Units.Radians.convertFrom(prefDrivetrain.turnSpeed.getValue(), Units.Degrees);
 
     // Requesting snapping ignores any previously calculated rotational speeds
     if (aimAtSpeaker.getAsBoolean()) {
       rVelocity = subDrivetrain.getVelocityToSnap(subDrivetrain.getAngleToSpeaker()).in(Units.RadiansPerSecond);
+
+      // im so sorry i cant come up with a better way to do this
+    } else if (north.getAsBoolean()) {
+      rVelocity = subDrivetrain.getVelocityToSnap(Units.Degrees.of(0 + northYaw.in(Units.Degrees)))
+          .in(Units.RadiansPerSecond);
+    } else if (east.getAsBoolean()) {
+      rVelocity = subDrivetrain.getVelocityToSnap(Units.Degrees.of(270 + northYaw.in(Units.Degrees)))
+          .in(Units.RadiansPerSecond);
+    } else if (south.getAsBoolean()) {
+      rVelocity = subDrivetrain.getVelocityToSnap(Units.Degrees.of(180 + northYaw.in(Units.Degrees)))
+          .in(Units.RadiansPerSecond);
+    } else if (west.getAsBoolean()) {
+      rVelocity = subDrivetrain.getVelocityToSnap(Units.Degrees.of(90 + northYaw.in(Units.Degrees)))
+          .in(Units.RadiansPerSecond);
+    } else if (chain.getAsBoolean()) {
+      rVelocity = subDrivetrain.getVelocityToChain().in(Units.RadiansPerSecond);
     }
```

Comment:
```suggestion
    } else if (north.getAsBoolean()) {
      rVelocity = rVelocity = subDrivetrain.getVelocityToSnap(northYaw.plus(Units.Degrees.of(0)));
```
rVelocity should be a velocity object to make all this math and code simpler. we should always convert at the end